src/NLog/Layouts/JsonLayout.cs

```diff
@@ -85,11 +99,71 @@ protected override string GetFormattedMessage(LogEventInfo logEvent)
             bool first = true;
             bool hasContent = false;
 
+            //Creates a copy of the list of specified attributes
+            List<JsonAttribute> attributes = new List<JsonAttribute>(this.Attributes);
+
+
+            //Adds attributes for all the properties
+            if (this.IncludeAllProperties)
+            {
+
+                foreach (var prop in logEvent.Properties)
+                {
+                    //Determine property name
+                    string propName = prop.Key.ToString();
+
+                    //Skips properties in the ExcludedProperties list
+                    if (this.ExcludedProperties.Contains(propName)) continue;
+
+                    //Create JsonAttribute based on type of prop.Value
+                    if (prop.Value == null)
+                    {
+                        attributes.Add(new JsonAttribute()
+                        {
+                            Name = propName,
+                            Encode = false, //Don't put quotes around null values
+                            Layout = "null"
+                        });
+                    }
+                    else if (
+                        prop.Value is bool
```

Comment:
Maybe we should integrate this with https://github.com/NLog/NLog/pull/1740  (injecting Json serializer) . What do you think?